httpx/status_codes.py

```diff
@@ -101,6 +103,7 @@ def is_server_error(cls, value: int) -> bool:
     UNSUPPORTED_MEDIA_TYPE = 415, "Unsupported Media Type"
     REQUESTED_RANGE_NOT_SATISFIABLE = 416, "Requested Range Not Satisfiable"
     EXPECTATION_FAILED = 417, "Expectation Failed"
+    I_AM_A_TEAPOT = 418, "I'm a teapot"
```

Comment:
At the risk of being too picky here, could we change the enum value to be `IM_A_TEAPOT`? :)

Reply:
I'm not opposed, but I did want to note that [requests uses both](https://github.com/kennethreitz/requests/blob/9a2e5df000691ad28613524ca7f80aa28a94a8d5/requests/status_codes.py#L72), and I took the constant name from the Symphony status constant on https://httpstatuses.com/418